README.md

```diff
@@ -1 +1,44 @@
-# oppia-ml
\ No newline at end of file
+# Oppia-ml
+
+Oppia-ml is a supplementary component which is used with Oppia for training machine learning models on a separate VM instance. Oppia puts job requests for training a classifier in database. Oppia-ml picks this job requests one by one, trains classifier for these requests and stores the result of training back into database. Oppia uses this stored results to predict outcome for answers.
+ 
+Oppia-ml is written in Python and uses various machine learning libraries for creating classifiers.
+
+## Installation
+
+### Installing on your development machine
+
+1. Clone this repo in folder oppia-ml.
+
+2. Open terminal and navigate to oppia-ml folder and run:
+  ```
+    git checkout develop
+    bash scripts/start.sh
+  ```
+
+ 
+### Deploying Oppia-ml on VM instance
+
+1. Clone this repo in oppia-ml folder of your VM instance.
+
+2. Install Supervisor on VM instance. Generally it can be installed by running simple pip command. You need superuser privileges to install it on VM. if this command does not work then follow instructions on official installation page. 
+  ```
+    pip install supervisor
+  ```
+
+3. Navigate to oppia-ml folder in terminal and run following commands:
+  ```
+    bash scripts/deploy.sh
+  ```
+
+4. Add shared secret key in VM and in Oppia for secure communication.
+  Shared key on VM is added using GCE metadata. Add two key - value pairs in metadata, one for “shared_secret_key” and other is “vm_id”. VM will automatically get the ID and secret from metadata.
```

Comment:
Consider using backticks instead so that the quoted bits display in `code font`.